### PR TITLE
Added an intermediate state during sync of locally processed events

### PR DESCRIPTION
**Pull request by rkodev**
Added an intermediate state during sync of locally processed events

#269

### CODE REVIEW

File: opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java [hunk 1]
```diff
@@ -1464,10 +1467,27 @@ public void addEvent(String baseEntityId, JSONObject jsonObject, String syncStat
...
+            Timber.e(e);
         }
     }
 
+    public void markEventAsProcessed(String formSubmissionId, String eventID) {
+        try {
+
+            ContentValues values = new ContentValues();
+            values.put(event_column.formSubmissionId.name(), formSubmissionId);
+            values.put(event_column.syncStatus.name(), StringUtils.isNotBlank(eventID) ? BaseRepository.TYPE_Synced : BaseRepository.TYPE_Unsynced);
```
Comment (githengi) [inline opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java]: Maybe you should update sync Status only if current sync status is  `TYPE_Unsynced`  or `TYPE_unprocessed` and not depend on checking if event has  eventId

**Comment by githengi** [comment]
Also with this change then if `ClientProcessor` is not called after saving event on Client Processor fails then these events will not be eligible to be picking by SyncIntent Service. I think you should amend SyncIntentService so that it picks events with status Uprocessed and Unsyced

File: opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java (hunk 1 above)
Comment (rkodev) [reply opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java]: Updated logic to only update records when the event was in a status of TYPE_unprocessed

Comment (rkodev) [reply opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java]: Resolved

**Comment by githengi** [comment]
Looks good add Javadoc documentation

**Pull request merged**